internal/services/storage/migration/account.go

```diff
@@ -204,3 +211,810 @@ func accountSchemaForV0AndV1() map[string]*pluginsdk.Schema {
+									Required:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
+								},
+
+								"netbios_domain_name": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"custom_domain": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+
+					"use_subdomain": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  false,
+					},
+				},
+			},
+		},
+
+		"enable_https_traffic_only": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"min_tls_version": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Default:  string(storage.TLS10),
+			ValidateFunc: validation.StringInSlice([]string{
+				string(storage.TLS10),
+				string(storage.TLS11),
+				string(storage.TLS12),
+			}, false),
+		},
+
+		"is_hns_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+			ForceNew: true,
+		},
+
+		"nfsv3_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+			ForceNew: true,
+		},
+
+		"allow_blob_public_access": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"shared_access_key_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"network_rules": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"bypass": {
+						Type:     pluginsdk.TypeSet,
+						Optional: true,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(storage.AzureServices),
+								string(storage.Logging),
+								string(storage.Metrics),
+								string(storage.None),
+							}, true),
+						},
+						Set: pluginsdk.HashString,
+					},
+
+					"ip_rules": {
+						Type:     pluginsdk.TypeSet,
+						Optional: true,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validate.StorageAccountIpRule,
+						},
+						Set: pluginsdk.HashString,
+					},
+
+					"virtual_network_subnet_ids": {
+						Type:     pluginsdk.TypeSet,
+						Optional: true,
+						Computed: true,
+						Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
+						Set:      pluginsdk.HashString,
+					},
+
+					"default_action": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(storage.DefaultActionAllow),
+							string(storage.DefaultActionDeny),
+						}, false),
```

Comment:
fwiw you don't need these for a state migration - only the Type and Required/Optional/Computed iirc

```suggestion
```